src/plugins/discover/public/application/view_components/canvas/index.tsx

```diff
@@ -152,20 +157,13 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history, optionalR
               timeFieldName={timeField}
             />
           )}
-          {fetchState.status === ResultStatus.ERROR && (
-            <DiscoverNoResults
-              queryString={data.query.queryString}
-              query={data.query.queryString.getQuery()}
-              savedQuery={data.query.savedQueries}
-              timeFieldName={timeField}
-            />
-          )}
           {fetchState.status === ResultStatus.UNINITIALIZED && (
             <DiscoverUninitialized onRefresh={() => refetch$.next()} />
           )}
           {fetchState.status === ResultStatus.LOADING && !rows?.length && <LoadingSpinner />}
           {(fetchState.status === ResultStatus.READY ||
-            (fetchState.status === ResultStatus.LOADING && !!rows?.length)) &&
+            (fetchState.status === ResultStatus.LOADING && !!rows?.length) ||
+            fetchState.status === ResultStatus.ERROR) &&
```

Comment:
if the first query returned an error and there are no previous query or previous rows, what does it show?

Reply:
Should we show no result UI here if the first query we execute is invalid? @ashwin-pc

Reply:
Yes it should show the last valid UI state